sw/device/lib/dif/dif_edn.c

```diff
@@ -4,15 +4,328 @@
+                               kMultiBitBool4True);
+  mmio_region_write32(edn->base_addr, EDN_CTRL_REG_OFFSET, reg);
+
+  return kDifOk;
+}
+
+dif_result_t dif_edn_get_status(const dif_edn_t *edn, dif_edn_status_t flag,
+                                bool *set) {
+  if (edn == NULL || set == NULL) {
+    return kDifBadArg;
+  }
+
+  uint32_t bit;
+  switch (flag) {
+    case kDifEdnStatusReady:
+      bit = EDN_SW_CMD_STS_CMD_RDY_BIT;
+      break;
+    case kDifEdnStatusCsrngAck:
+      bit = EDN_SW_CMD_STS_CMD_STS_BIT;
+      break;
+    default:
+      return kDifBadArg;
+  }
+
+  uint32_t reg = mmio_region_read32(edn->base_addr, EDN_SW_CMD_STS_REG_OFFSET);
+  *set = bitfield_bit32_read(reg, bit);
+  return kDifOk;
+}
+
+dif_result_t dif_edn_get_errors(const dif_edn_t *edn, uint32_t *unhealthy_fifos,
+                                uint32_t *errors) {
+  if (edn == NULL || unhealthy_fifos == NULL || errors == NULL) {
+    return kDifBadArg;
+  }
+  *unhealthy_fifos = 0;
+  *errors = 0;
+
+  uint32_t reg = mmio_region_read32(edn->base_addr, EDN_ERR_CODE_REG_OFFSET);
+  *unhealthy_fifos =
+      bitfield_bit32_copy(*unhealthy_fifos, kDifEdnFifoReseedCmd, reg,
+                          EDN_ERR_CODE_SFIFO_RESCMD_ERR_BIT);
+  *unhealthy_fifos =
+      bitfield_bit32_copy(*unhealthy_fifos, kDifEdnFifoGenerateCmd, reg,
+                          EDN_ERR_CODE_SFIFO_GENCMD_ERR_BIT);
+
+  *errors = bitfield_bit32_copy(*errors, kDifEdnErrorAckSm, reg,
+                                EDN_ERR_CODE_EDN_ACK_SM_ERR_BIT);
+  *errors = bitfield_bit32_copy(*errors, kDifEdnErrorMainSm, reg,
+                                EDN_ERR_CODE_EDN_MAIN_SM_ERR_BIT);
+  *errors = bitfield_bit32_copy(*errors, kDifEdnErrorCounterFault, reg,
+                                EDN_ERR_CODE_EDN_CNTR_ERR_BIT);
+  *errors = bitfield_bit32_copy(*errors, kDifEdnErrorFifoWrite, reg,
+                                EDN_ERR_CODE_FIFO_WRITE_ERR_BIT);
+  *errors = bitfield_bit32_copy(*errors, kDifEdnErrorFifoRead, reg,
+                                EDN_ERR_CODE_FIFO_READ_ERR_BIT);
+  *errors = bitfield_bit32_copy(*errors, kDifEdnErrorFifoFullAndEmpty, reg,
+                                EDN_ERR_CODE_FIFO_STATE_ERR_BIT);
+
+  return kDifOk;
+}
+
+dif_result_t dif_edn_get_cmd_force_unhealthy_fifo(const dif_edn_t *edn,
+                                                  dif_edn_fifo_t fifo) {
+  if (edn == NULL) {
+    return kDifBadArg;
+  }
+
+  uint32_t fifo_bit;
+  switch (fifo) {
+    case kDifEdnFifoReseedCmd:
+      fifo_bit = EDN_ERR_CODE_SFIFO_RESCMD_ERR_BIT;
+      break;
+    case kDifEdnFifoGenerateCmd:
+      fifo_bit = EDN_ERR_CODE_SFIFO_GENCMD_ERR_BIT;
+    default:
+      return kDifBadArg;
+  }
+
+  DIF_RETURN_IF_ERROR(check_locked(edn));
+  mmio_region_write32(edn->base_addr, EDN_ERR_CODE_TEST_REG_OFFSET, fifo_bit);
+  return kDifOk;
+}
+
+dif_result_t dif_edn_get_cmd_force_error(const dif_edn_t *edn,
```

Comment:
As mentioned above, I think the name `dif_edn_error_force()` may be more consistent with the autogenerated `dif_edn_{alert,irq}_force()` DIFs. wdyt?

Reply:
Sure, we can do that.